### PR TITLE
Validation skipped fix

### PR DESCRIPTION
**Pull request by br648**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

Added a new method to distinguish between active and staged jobs.

Fixes https://github.com/ibi-group/datatools-server/issues/395

### CODE REVIEW

File: src/main/java/com/conveyal/datatools/manager/utils/JobUtils.java [hunk 1]
```diff
@@ -98,23 +98,39 @@ public static MonitorableJob getJobById(String userId, String jobId, boolean cle
...
         return jobsStillActive;
     }
 
+    /**
+     * Filter jobs based on status. Once a job has been created the status values are available for review. A staged job
+     * may or may not be active.
+     */
+    public static Set<MonitorableJob> filterStagedJobs(Set<MonitorableJob> jobs) {
+        // Note: this must be a thread-safe set in case it is placed into the DataManager#userJobsMap.
+        Set<MonitorableJob> jobsStillActive = Sets.newConcurrentHashSet();
```
Comment (binh-dam-ibigroup) [inline src/main/java/com/conveyal/datatools/manager/utils/JobUtils.java:127]: Rename to `stagedJobs` or something similar.

**Review by binh-dam-ibigroup — APPROVED** [review]
LGTM, with a recommendation to rename a variable. Thanks @br648 for the changes!

File: src/main/java/com/conveyal/datatools/manager/utils/JobUtils.java (hunk 1 above)
Comment (br648) [reply src/main/java/com/conveyal/datatools/manager/utils/JobUtils.java]: Updated.

**Review by landonreed — CHANGES_REQUESTED** [review]
Just a couple of changes in #397 to dry things out.

**Review by landonreed — APPROVED** [review]

File: src/main/java/com/conveyal/datatools/manager/models/FeedSource.java [hunk 2]
```diff
@@ -615,7 +615,7 @@ public FeedSource clone () throws CloneNotSupportedException {
      * occurring at the same time).
      */
     public boolean hasJobsInProgress() {
-        return JobUtils.filterActiveJobs(JobUtils.getAllJobs()).stream().anyMatch(job -> {
+        return JobUtils.filterStagedAndActiveJobs(JobUtils.getAllJobs()).stream().anyMatch(job -> {
```
Comment (evansiroky) [inline src/main/java/com/conveyal/datatools/manager/models/FeedSource.java:618]: This creates a regression from https://github.com/ibi-group/datatools-server/commit/db462d05dd7622ef6f0ff03769350ce2e9622347, which I suppose created the other regression that this PR aims to fix. This changed filtering of jobs now filters out actively running recurring jobs which have already "completed" at least once. I added a comment about this, but it's easy to miss this nuance (see below) https://github.com/ibi-group/datatools-server/blob/ddb3cc913525fe93598fdd870b86fded5127f995/src/main/java/com/conveyal/datatools/common/status/MonitorableJob.java#L33-L37

Therefore, I think if we change this to the following, it should work again for auto-deployment purposes.

```suggestion
        return JobUtils.getAllActiveJobs().stream().anyMatch(job -> {
```

**Review by evansiroky — CHANGES_REQUESTED** [review]
See one blocking comment.

**Review by evansiroky — APPROVED** [review]

**Pull request merged**